keras/engine/training.py

```diff
@@ -1091,8 +1092,12 @@ def compute_metrics(self, x, y, y_pred, sample_weight):
         del x  # The default implementation does not use `x`.
         self.compiled_metrics.update_state(y, y_pred, sample_weight)
         # Collect metrics to return
+        return self._compute_metrics_result(self.metrics)
+
+    def _compute_metrics_result(self, metrics):
```

Comment:
Please explain the need for this need method

Reply:
It is meant to be used here https://github.com/keras-team/keras/blob/3545340d24f1fb7bf5e3d1e2ee860a09a1598eeb/keras/engine/training.py#L1579

to force metrics to be computed at the epoch level.

This allows users to prevent computing `Metric.result` on the batch level, if for example it is computationally intensive. They can customize `compute_metrics` as follow:

```python
# User defined `compute_metrics`:
def compute_metrics(...):
    # Only perform state updates
    [m.update_state(...) for m in self.metrics]
    return {} # this is ok, Metric.result will be called on the epoch level
```